R/dependencies.R

```diff
@@ -122,6 +122,27 @@ dirDependencies <- function(dir) {
   packratDirRegex <- "(?:^|/)packrat"
   R_files <- grep(packratDirRegex, R_files, invert = TRUE, value = TRUE)
 
+  ## Avoid anything on the list of ignored directories
+  ignoredDir <- get_opts("ignored.directories")
+  if (length(ignoredDir) > 0) {
+    # Make sure all the directories end with a slash...
+    ignoredDir <- ifelse(
+      substr(ignoredDir, nchar(ignoredDir), nchar(ignoredDir)) != "/",
+      paste0(ignoredDir, "/"),
+      ignoredDir
+    )
+
+    # Make a regex to match any of them.
+    ignoredDirRegex <- paste0(
+      "(?:^",
+      paste0(
+        ignoredDir,
```

Comment:
Should we escape special characters here? (E.g. if a user wanted to ignore a directory called `data.old`, we might need to escape that as `data\.old`)

Reply:
Good point, it should be escaped! Fortunately most of the regex special characters aren't very common in filenames, other than ".".

Without escapes, "data.old" will still match "data.old", but also incorrectly matches e.g. "dataFold"